pandas/formats/format.py

```diff
@@ -2042,7 +2042,13 @@ def get_result_as_array(self):
         """
 
         if self.formatter is not None:
-            return np.array([self.formatter(x) for x in self.values])
+            # na_rep takes precedence over the specified formatter
+            mask = isnull(self.values)
+            formatted_values = np.array([self.formatter(x)
+                                         for x in self.values])
+            if self.na_rep is not None:
```

Comment:
Why? IIUC and we're trying to get `na_rep` to have precedence over `formatter` (and `float_format`) iterating only over non-null values would be the wrong thing to do.